internal/provider/resource_maas_server.go

```diff
@@ -0,0 +1,62 @@
+package provider
+
```

Comment:
Is the MAAS server the region controller? I will continue all the reviews, but we may need to rename this to align the elements. We can chat about it, no need to do it now before we talk.